src/ui/public/utils/decode_geo_hash.js

```diff
@@ -56,3 +56,17 @@ function refine_interval(interval, cd, mask) {
 }
 
 export default decodeGeoHash;
+
+/**
+ *
+ * @param {number} precision the geohash precision (1<=precision<=12).
+ * @param {number} axis constant for the axis 0=lengthwise (ie. columns, along longitude), 1=heightwise (ie. rows, along latitude)
+ * @returns {number} Number of geohash cells at that precision
+ */
+export function geohashCells(precision, axis) {
+  let cells = 1;
+  for (let i = 1; i <= precision; i += 1) {
+    cells *= (i % 2 === axis) ? 4 : 8;
```

Comment:
These aren't "magic" in the sense that they map to alternation-pattern of geohash levels, which subdivide alternately between 4 and 8 depending on even/oddness of precision and lengthwise/heightwise.

I'll make this more clear.